src/Avalonia.Layout/ElementManager.cs

```diff
@@ -93,12 +93,12 @@ public ILayoutable GetAt(int realizedIndex)
             else
             {
                 // realizedIndex and dataIndex are the same (everything is realized)
-                element = _context.GetOrCreateElementAt(
+                element = _context!.GetOrCreateElementAt(
                     realizedIndex,
                     ElementRealizationOptions.ForceCreate | ElementRealizationOptions.SuppressAutoRecycle);
             }
 
-            return element;
+            return element!;
```

Comment:
It doesn't feel right that "element" can be null here.
It must have value in all branches, and I think Roslyn is not that silly.

Maybe you can remove "?" from line 74 and it will just work.

Reply:
Unfortunately not, Roslyn isn't that intelligent yet:

![image](https://user-images.githubusercontent.com/1775141/146695426-e8d706ca-65d6-4b6c-8954-82f4ada87031.png)

Reply:
Thought I guess it might be better if the `!` was on line 90.

Reply:
it's probably better to read this indexer once before condition to make roslyn happy (and we will win couple of nanoseconds by avoiding additional memory read)

Reply:
Why not just assign it to `element` first and only do the null check after (remove the else)?

```C#
element = _realizedElements[realizedIndex]
if (element == null)
{
   ...
}
```

Reply:
@Symbai yeah that would probably make more sense. Code is from [WinUI](https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/Repeater/ElementManager.cpp#L50-L75) and it's written like that there, but I think it would make sense to diverge here.